### PR TITLE
[15.0][IMP] web_notify: action button name and close

### PR DESCRIPTION
**Pull request by chienandalu**
- We can now set a button name to the notification action.
- We can set an icon button as well.
- After the button is clicked, the notification gets closed.

image

cc @Tecnativa TT47134

take a look @IT-Ideas @rousseldenis 

ping @sergio-teruel

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]
Great!!

**Comment by chienandalu** [comment]
could you review @carlosdauden @CarlosRoca13 ?

**Review by rousseldenis — APPROVED** [review]
LGTM

**Review by CarlosRoca13 — APPROVED** [review]

File: web_notify/README.rst [hunk 1]
```diff
@@ -91,16 +91,18 @@ The notifications can bring interactivity with some buttons.
...
+give to our button with the ``button_name`` key in the action context `params` key:
 
 .. code-block:: python
 
-    action = self.env["ir.actions.act_window"]._for_xml_id('sale.action_orders')
-    action.update({
-        'res_id': self.id,
-        'views': [(False, 'form')],
-    })
+   action = self.env["ir.actions.act_window"]._for_xml_id('sale.action_orders')
```
Comment (pedrobaeza) [inline web_notify/README.rst]: You have reduced here one space of indentation, which is not really correct and increases the diff.

File: web_notify/readme/USAGE.rst [hunk 2]
```diff
@@ -37,26 +37,28 @@ The notifications can bring interactivity with some buttons.
...
+give to our button with the ``button_name`` key in the action context `params` key:
 
 .. code-block:: python
 
-    action = self.env["ir.actions.act_window"]._for_xml_id('sale.action_orders')
-    action.update({
-        'res_id': self.id,
-        'views': [(False, 'form')],
-    })
+   action = self.env["ir.actions.act_window"]._for_xml_id('sale.action_orders')
```
Comment (pedrobaeza) [inline web_notify/readme/USAGE.rst]: The same here.

File: web_notify/readme/USAGE.rst [hunk 3]
```diff
@@ -37,26 +37,28 @@ The notifications can bring interactivity with some buttons.
...
-.. figure:: static/description/notifications_screenshot.gif
+.. figure:: ../static/description/notifications_screenshot.gif
    :scale: 80 %
    :alt: Sample notifications
 
 You can test the behaviour of the notifications by installing this module in a demo database.
 Access the users form through Settings -> Users & Companies. You'll see a tab called "Test web notify", here you'll find two buttons that'll allow you test the module.
 
-.. figure:: static/description/test_notifications_demo.png
+.. figure:: ../static/description/test_notifications_demo.png
```
Comment (pedrobaeza) [inline web_notify/readme/USAGE.rst]: Try to put the image on the `img` folder:

```suggestion
.. figure:: ../static/img/test_notifications_demo.png
```

and rebuild the README (forcing it).

**Comment by chienandalu** [comment]
Changes done, let's see if merge bot is happy now (not very confident thou)

**Comment by chienandalu** [comment]
it failed as well :confounded:

**Comment by pedrobaeza** [comment]
I think the GIF is in a format that PIL doesn't load it. Can you reprocess it?

**Comment by chienandalu** [comment]
Changed to png. Let's try again

**Comment by pedrobaeza** [comment]
But we are losing the animation then... I was just talking about opening the GIF and saving again with a software that allows it to reprocess the GIF to remove any offending bit.

**Comment by ap-wtioit** [comment]
Not sure, but wanted to give a hint, this looks like https://github.com/pypa/readme_renderer/issues/304 from the readme-renderer here: https://github.com/OCA/oca-github-bot/blob/master/requirements.txt#L48C8-L48C16

**Comment by chienandalu** [comment]
Thanks @ap-wtioit ! I just pushed changes that avoid scaling those screenshots. Hopefully it will be ok now

**Pull request merged**